### PR TITLE
Unconditionally set `RUN_QUEUED`

### PR DESCRIPTION
**Pull request by bugadani**
This PR sets `RUN_QUEUED` unconditionally, which results in a tiny perf win:

- ESP32-S3 (atomic CAS): `16652/100` to `15849/100` cycles.
- ESP32-S2 (critical section): `26641/100` to `25815/100` cycles.

Based on #3628

### CODE REVIEW

**Comment by bugadani** [comment]
The test cases fail on main, because exited tasks are not pended there:

```
---- waking_with_old_waker_after_respawn stdout ----
thread 'waking_with_old_waker_after_respawn' panicked at tests\test.rs:230:5:
assertion `left == right` failed
  left: ["pend", "yield_now", "pend", "poll task1"]
 right: ["pend", "yield_now", "pend", "poll task1", "pend"]
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- waking_after_completion_does_not_poll stdout ----
thread 'waking_after_completion_does_not_poll' panicked at tests\test.rs:176:5:
assertion `left == right` failed
  left: ["pend", "poll task1", "pend", "poll task1"]
 right: ["pend", "poll task1", "pend", "pend", "pend", "poll task1"]
```

Pending exited tasks make sure the RUNNING bit (which we now unconditionally set) is cleared, which is needed to not leak a task's storage. This clearing is done by `run_dequeue` and I think it does not matter which executor does it - but the CLAIMED state makes sure it's the old one pre-respawn, and the new one post-respawn. During respawn, pending isn't necessary so we don't do it.

**Comment by bugadani** [comment]
I've walked back on the CLAIMED state. It's not necessary because spawning a task sets RUN_QUEUED, which in turn prevents run-enqueueing a task in the wrong executor already.

File: embassy-executor/src/raw/state_atomics_arm.rs [hunk 1]
```diff
@@ -72,7 +72,7 @@ impl State {
                 let state: u32;
                 asm!("ldrex {}, [{}]", out(reg) state, in(reg) self, options(nostack));
 
-                if (state & STATE_RUN_QUEUED != 0) || (state & STATE_SPAWNED == 0) {
+                if state & STATE_RUN_QUEUED != 0 {
```
[the remark this answers is not in the document]
Comment (Dirbaio) [reply embassy-executor/src/raw/state_atomics_arm.rs:75]: yep!

[the remark this answers is not in the document]
Comment (Dirbaio) [reply embassy-executor/src/raw/state_atomics_arm.rs:75]: well, or swap with true. not sure which one is faster or if the compiler is smart enough to optimize them to the same.

**Pull request merged**